chainer/functions/gaussian.py

```diff
@@ -0,0 +1,73 @@
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class Gaussian(function.Function):
+
+    """Gaussian sampling function.
+
+    It takes mean and log(variance) as input and output sample drawn from
+    gaussian N(mean, variance).
+
+    """
+
+    def __init__(self):
+        self.eps = None
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+
+        m_type, v_type = in_types
+        type_check.expect(
+            m_type.dtype == numpy.float32,
+            v_type.dtype == numpy.float32,
+            m_type.shape == v_type.shape,
+        )
+
+    def check_type_fbackward(self, in_types, out_types):
+        type_check.expect(out_types.size() == 1)
+        m_type, v_type = in_types
+        g_type, = out_types
+
+        type_check.expect(
+            g_type.dtype == numpy.float32,
+            g_type.shape == m_type.shape,
+        )
+
+    def check_type_backward(self, in_types, out_types):
+        type_check.expect(out_types.size() == 1)
+
+    def forward_cpu(self, inputs):
+        mean, ln_var = inputs
+        if self.eps is None:
```

Comment:
Unless `self.eps` is assigned directly, self.eps is always `None`. Did you intend to add eps to the argument of `__init__` ?